iceoryx_posh/source/roudi/roudi_process.cpp

```diff
@@ -552,6 +527,101 @@ void ProcessManager::addSenderForProcess(const ProcessName_t& name,
     }
 }
 
+void ProcessManager::addSubscriberForProcess(const ProcessName_t& name,
+                                             const capro::ServiceDescription& service,
+                                             const uint64_t& historyRequest,
+                                             const RunnableName_t& runnable,
+                                             const PortConfigInfo& portConfigInfo) noexcept
+{
+    std::lock_guard<std::mutex> g(m_mutex);
+
+    RouDiProcess* process = getProcessFromList(name);
+    if (nullptr != process)
+    {
+        // create a SubscriberPort
+
+        /// @todo: it might be useful to encapsulate this into some kind of port factory
+        /// (which maybe contains a m_shmMgr)
+        /// main goal would be to isolate the port creation logic as it becomes more complex
+        /// pursuing this further could lead to a separate management entity for ports
+        /// which could support queries like: find all ports with a given service or some other
+        /// specific attribute (to allow efficient and well encapsulated lookup)
+
+        auto maybeSubscriber =
+            m_portManager.acquireSubscriberPortData(service, historyRequest, name, runnable, portConfigInfo);
+
+        if (!maybeSubscriber.has_error())
+        {
+            // send ReceiverPort to app as a serialized relative pointer
```

Comment:
copy & waste ;-) The term "Receiver" is deprecated

Reply:
missed